File: app/parameter_profiles.py
"""
Parameter Profiles for Free Thinkers
Contains predefined and customizable parameter profiles for different use cases
"""

import json
import os
from pathlib import Path

# Define the directory for storing parameter profiles
PROFILES_DIR = Path(os.path.expanduser("~/.freethinkers/parameter_profiles/"))

# Initialize with default parameter profiles
DEFAULT_PARAMETER_PROFILES = {
    "general": [
        {
            "name": "Balanced",
            "description": "General-purpose settings with balanced creativity and reliability",
            "parameters": {
                "temperature": 0.7,
                "top_p": 0.95,
                "top_k": 40,
                "repetition_penalty": 1.1,
                "context_window": 2048
            }
        },
        {
            "name": "Creative",
            "description": "Optimized for creative and diverse outputs",
            "parameters": {
                "temperature": 0.85,
                "top_p": 0.92,
                "top_k": 50,
                "repetition_penalty": 1.05,
                "context_window": 2048
            }
        },
        {
            "name": "Precise",
            "description": "Optimized for factual accuracy and consistent outputs",
            "parameters": {
                "temperature": 0.3,
                "top_p": 0.85,
                "top_k": 20,
                "repetition_penalty": 1.2,
                "context_window": 2048
            }
        }
    ],
    "task_specific": [
        {
            "name": "Creative Writing",
            "description": "Optimized for storytelling, fiction, and creative content",
            "parameters": {
                "temperature": 0.8,
                "top_p": 0.92,
                "top_k": 50,
                "repetition_penalty": 1.05,
                "context_window": 2048
            }
        },
        {
            "name": "Factual Q&A",
            "description": "Optimized for answering factual questions accurately",
            "parameters": {
                "temperature": 0.3,
                "top_p": 0.85,
                "top_k": 10,
                "repetition_penalty": 1.15,
                "context_window": 2048
            }
        },
        {
            "name": "Code Generation",
            "description": "Optimized for generating reliable code",
            "parameters": {
                "temperature": 0.4,
                "top_p": 0.95,
                "top_k": 30,
                "repetition_penalty": 1.1,
                "context_window": 2048
            }
        },
        {
            "name": "Brainstorming",
            "description": "Maximizes diversity and creativity for idea generation",
            "parameters": {
                "temperature": 0.9,
                "top_p": 0.98,
                "top_k": 60,
                "repetition_penalty": 1.0,
                "context_window": 2048
            }
        },
        {
            "name": "Analytical",
            "description": "Optimized for analytical reasoning and structured responses",
            "parameters": {
                "temperature": 0.5,
                "top_p": 0.9,
                "top_k": 30,
                "repetition_penalty": 1.1,
                "context_window": 2048
            }
        }
    ],
    "model_specific": {
        "mistral-7b": {
            "Balanced": {
                "temperature": 0.7,
                "top_p": 0.95,
                "top_k": 40,
                "repetition_penalty": 1.1
            },
            "Creative": {
                "temperature": 0.9,
                "top_p": 0.95,
                "top_k": 50,
                "repetition_penalty": 1.05
            },
            "Precise": {
                "temperature": 0.3,
                "top_p": 0.85,
                "top_k": 20,
                "repetition_penalty": 1.2
            },
            "Creative Writing": {
                "temperature": 0.85,
                "top_p": 0.92,
                "top_k": 50,
                "repetition_penalty": 1.05
            },
            "Factual Q&A": {
                "temperature": 0.25,
                "top_p": 0.85,
                "top_k": 10,
                "repetition_penalty": 1.15
            }
        },
        "llama3.2": {
            "Balanced": {
                "temperature": 0.65,
                "top_p": 0.9,
                "top_k": 35,
                "repetition_penalty": 1.1
            },
            "Creative": {
                "temperature": 0.8,
                "top_p": 0.92,
                "top_k": 45,
                "repetition_penalty": 1.05
            },
            "Precise": {
                "temperature": 0.25,
                "top_p": 0.8,
                "top_k": 20,
                "repetition_penalty": 1.2
            },
            "Code Generation": {
                "temperature": 0.3,
                "top_p": 0.95,
                "top_k": 25,
                "repetition_penalty": 1.1
            }
        },
        # gemma-2b-it removed from model-specific profiles
        "llava-phi3:latest": {
            "Balanced": {
                "temperature": 0.7,
                "top_p": 0.95,
                "top_k": 40,
                "repetition_penalty": 1.1
            },
            "Creative": {
                "temperature": 0.8,
                "top_p": 0.92,
                "top_k": 45,
                "repetition_penalty": 1.05
            },
            "Precise": {
                "temperature": 0.4,
                "top_p": 0.9,
                "top_k": 25,
                "repetition_penalty": 1.15
            }
        }
    },
    "custom": []
}

def initialize_profiles():
    """Initialize parameter profiles directory and create default profiles file."""
    try:
        PROFILES_DIR.mkdir(parents=True, exist_ok=True)
        profiles_file = PROFILES_DIR / "profiles.json"
        
        # Create default profiles file if it doesn't exist
        if not profiles_file.exists():
            with open(profiles_file, 'w') as f:
                json.dump(DEFAULT_PARAMETER_PROFILES, f, indent=2)
            return DEFAULT_PARAMETER_PROFILES
        else:
            # Read existing profiles
            with open(profiles_file, 'r') as f:
                return json.load(f)
    except Exception as e:
        print(f"Error initializing parameter profiles: {str(e)}")
        return DEFAULT_PARAMETER_PROFILES

def get_profiles():
    """Get all parameter profiles."""
    try:
        profiles_file = PROFILES_DIR / "profiles.json"
        if profiles_file.exists():
            with open(profiles_file, 'r') as f:
                return json.load(f)
        else:
            return initialize_profiles()
    except Exception as e:
        print(f"Error loading parameter profiles: {str(e)}")
        return DEFAULT_PARAMETER_PROFILES

def get_profiles_for_model(model_name):
    """Get parameter profiles for a specific model."""
    profiles = get_profiles()
    
    # Start with general profiles
    result = {profile["name"]: profile["parameters"] for profile in profiles["general"]}
    
    # Add task specific profiles
    for profile in profiles["task_specific"]:
        result[profile["name"]] = profile["parameters"]
    
    # Add model-specific overrides if they exist
    if model_name in profiles["model_specific"]:
        model_profiles = profiles["model_specific"][model_name]
        for profile_name, parameters in model_profiles.items():
            result[profile_name] = parameters
    
    # Add custom profiles
    for profile in profiles["custom"]:
        if "model" not in profile or profile["model"] == model_name:
            result[profile["name"]] = profile["parameters"]
    
    return result

def add_custom_profile(name, description, parameters, model=None):
    """Add a new custom parameter profile."""
    try:
        profiles = get_profiles()
        
        # Create new profile
        new_profile = {
            "name": name,
            "description": description,
            "parameters": parameters
        }
        
        # Add model specificity if provided
        if model:
            new_profile["model"] = model
        
        # Check for duplicate
        for idx, profile in enumerate(profiles["custom"]):
            if profile["name"] == name and ("model" not in profile or profile.get("model") == model):
                # Replace existing profile
                profiles["custom"][idx] = new_profile
                break
        else:
            # Add new profile
            profiles["custom"].append(new_profile)
        
        # Save profiles
        profiles_file = PROFILES_DIR / "profiles.json"
        with open(profiles_file, 'w') as f:
            json.dump(profiles, f, indent=2)
        
        return True, "Profile saved successfully"
    except Exception as e:
        return False, f"Error saving profile: {str(e)}"

def delete_custom_profile(name, model=None):
    """Delete a custom parameter profile."""
    try:
        profiles = get_profiles()
        
        # Filter out the profile to delete
        profiles["custom"] = [
            p for p in profiles["custom"] 
            if p["name"] != name or (model and p.get("model") != model)
        ]
        
        # Save profiles
        profiles_file = PROFILES_DIR / "profiles.json"
        with open(profiles_file, 'w') as f:
            json.dump(profiles, f, indent=2)
        
        return True, "Profile deleted successfully"
    except Exception as e:
        return False, f"Error deleting profile: {str(e)}"

def get_profile(name, model_name=None):
    """Get a specific parameter profile by name."""
    profiles = get_profiles()
    
    # Check model-specific profiles first
    if model_name and model_name in profiles["model_specific"]:
        model_profiles = profiles["model_specific"][model_name]
        if name in model_profiles:
            return model_profiles[name]
    
    # Check custom profiles
    for profile in profiles["custom"]:
        if profile["name"] == name and (not model_name or "model" not in profile or profile.get("model") == model_name):
            return profile["parameters"]
    
    # Check task-specific profiles
    for profile in profiles["task_specific"]:
        if profile["name"] == name:
            return profile["parameters"]
    
    # Check general profiles
    for profile in profiles["general"]:
        if profile["name"] == name:
            return profile["parameters"]
    
    # Return default parameters if profile not found
    return {
        "temperature": 0.7,
        "top_p": 0.95,
        "top_k": 40,
        "repetition_penalty": 1.1,
        "context_window": 2048
    }

# Initialize profiles on module import
initialize_profiles()

File: static/components/parameter_profiles.js
/**
 * Parameter Profiles Management
 * Handles loading, saving, and applying parameter profiles
 */

class ParameterProfiles {
    constructor() {
        this.profiles = {};
        this.currentModel = null;
        this.currentProfile = "Balanced";
    }

    /**
     * Initialize the parameter profiles system
     * @param {string} modelName - The current model name
     */
    async init(modelName) {
        try {
            this.currentModel = modelName || "llama3.2";
            
            // Load profiles for current model
            await this.loadProfilesForModel(this.currentModel);
            
            // Initialize UI components
            this.initProfilesUI();
            
            console.log(`Parameter profiles initialized for model: ${this.currentModel}`);
            return true;
        } catch (error) {
            console.error(`Error initializing parameter profiles: ${error}`);
            return false;
        }
    }
    
    /**
     * Load parameter profiles for a specific model
     * @param {string} modelName - The model name
     */
    async loadProfilesForModel(modelName) {
        try {
            const response = await fetch(`/api/parameter_profiles/model/${modelName}`);
            if (response.ok) {
                this.profiles = await response.json();
                return this.profiles;
            } else {
                console.error(`Error loading profiles: ${response.statusText}`);
                return {};
            }
        } catch (error) {
            console.error(`Error loading profiles: ${error}`);
            return {};
        }
    }
    
    /**
     * Initialize the profiles UI components
     */
    initProfilesUI() {
        // Create profiles dropdown in parameter controls sidebar
        const paramContent = document.querySelector('.parameter-controls-content');
        if (!paramContent) return;
        
        // Create profiles section if it doesn't exist
        let profilesSection = document.querySelector('.profiles-section');
        if (!profilesSection) {
            profilesSection = document.createElement('div');
            profilesSection.className = 'profiles-section mb-4';
            profilesSection.innerHTML = `
                <div class="section-header">
                    <h4 class="mb-2">Parameter Profiles</h4>
                    <p class="text-muted small mb-0">Save and apply different parameter configurations</p>
                </div>
                <div class="profile-selector mb-3">
                    <label for="profileSelect" class="form-label">Select Profile:</label>
                    <select id="profileSelect" class="form-select"></select>
                </div>
                <div class="profile-description mb-3" id="profileDescription">
                    <p class="text-muted">Select a profile to view its description</p>
                </div>
                <div class="profile-actions mb-3">
                    <button id="applyProfileBtn" class="btn btn-primary me-2">
                        <i class="fas fa-check"></i> Apply Profile
                    </button>
                    <button id="saveAsNewBtn" class="btn btn-outline-secondary">
                        <i class="fas fa-save"></i> Save Current as New
                    </button>
                </div>
                <div id="saveProfileModal" class="modal fade" tabindex="-1">
                    <div class="modal-dialog">
                        <div class="modal-content">
                            <div class="modal-header">
                                <h5 class="modal-title">Save Parameter Profile</h5>
                                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
                            </div>
                            <div class="modal-body">
                                <div class="mb-3">
                                    <label for="profileName" class="form-label">Profile Name</label>
                                    <input type="text" class="form-control" id="profileName" placeholder="My Custom Profile">
                                </div>
                                <div class="mb-3">
                                    <label for="profileDescription" class="form-label">Description</label>
                                    <textarea class="form-control" id="profileDescription" rows="3" placeholder="Describe what this profile is optimized for..."></textarea>
                                </div>
                                <div class="form-check mb-3">
                                    <input class="form-check-input" type="checkbox" id="modelSpecificCheck">
                                    <label class="form-check-label" for="modelSpecificCheck">
                                        Make this profile specific to current model
                                    </label>
                                </div>
                            </div>
                            <div class="modal-footer">
                                <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Cancel</button>
                                <button type="button" class="btn btn-primary" id="saveProfileBtn">Save Profile</button>
                            </div>
                        </div>
                    </div>
                </div>
            `;
            
            // Insert at the beginning of the parameter controls
            paramContent.insertBefore(profilesSection, paramContent.firstChild);
            
            // Initialize the dropdown with profiles
            this.updateProfilesDropdown();
            
            // Add event listeners
            this.addProfileEventListeners();
        } else {
            // Just update the dropdown
            this.updateProfilesDropdown();
        }
    }
    
    /**
     * Update the profiles dropdown with current profiles
     */
    updateProfilesDropdown() {
        const profileSelect = document.getElementById('profileSelect');
        if (!profileSelect) return;
        
        // Clear existing options
        profileSelect.innerHTML = '';
        
        // Add profile options
        const profileNames = Object.keys(this.profiles);
        profileNames.sort(); // Sort alphabetically
        
        // Group profiles
        const generalProfiles = ['Balanced', 'Creative', 'Precise'];
        const taskProfiles = ['Creative Writing', 'Factual Q&A', 'Code Generation', 'Brainstorming', 'Analytical'];
        const customProfiles = profileNames.filter(name => 
            !generalProfiles.includes(name) && !taskProfiles.includes(name));
        
        // Add general profiles
        const generalGroup = document.createElement('optgroup');
        generalGroup.label = 'General';
        generalProfiles.forEach(name => {
            if (this.profiles[name]) {
                const option = document.createElement('option');
                option.value = name;
                option.textContent = name;
                if (name === this.currentProfile) {
                    option.selected = true;
                }
                generalGroup.appendChild(option);
            }
        });
        profileSelect.appendChild(generalGroup);
        
        // Add task-specific profiles
        const taskGroup = document.createElement('optgroup');
        taskGroup.label = 'Task-Specific';
        taskProfiles.forEach(name => {
            if (this.profiles[name]) {
                const option = document.createElement('option');
                option.value = name;
                option.textContent = name;
                if (name === this.currentProfile) {
                    option.selected = true;
                }
                taskGroup.appendChild(option);
            }
        });
        if (taskGroup.children.length > 0) {
            profileSelect.appendChild(taskGroup);
        }
        
        // Add custom profiles
        if (customProfiles.length > 0) {
            const customGroup = document.createElement('optgroup');
            customGroup.label = 'Custom';
            customProfiles.forEach(name => {
                const option = document.createElement('option');
                option.value = name;
                option.textContent = name;
                if (name === this.currentProfile) {
                    option.selected = true;
                }
                customGroup.appendChild(option);
            });
            profileSelect.appendChild(customGroup);
        }
        
        // Update description for current profile
        this.updateProfileDescription();
        
        // Update parameters if a profile is selected
        if (profileSelect.value) {
            this.currentProfile = profileSelect.value;
        }
    }
    
    /**
     * Update the profile description panel
     */
    updateProfileDescription() {
        const profileSelect = document.getElementById('profileSelect');
        const descriptionElement = document.getElementById('profileDescription');
        if (!profileSelect || !descriptionElement) return;
        
        const selectedProfile = profileSelect.value;
        
        // Set description based on profile
        let description = '';
        
        switch (selectedProfile) {
            case 'Balanced':
                description = 'General-purpose settings with balanced creativity and reliability.';
                break;
            case 'Creative':
                description = 'Higher temperature and diversity for more creative outputs.';
                break;
            case 'Precise':
                description = 'Lower temperature for more deterministic, factual responses.';
                break;
            case 'Creative Writing':
                description = 'Optimized for storytelling, fiction, and creative content.';
                break;
            case 'Factual Q&A':
                description = 'Optimized for answering factual questions accurately.';
                break;
            case 'Code Generation':
                description = 'Settings tuned for generating reliable and correct code.';
                break;
            case 'Brainstorming':
                description = 'Maximizes diversity for idea generation and exploratory thinking.';
                break;
            case 'Analytical':
                description = 'Balanced settings for analytical reasoning and structured responses.';
                break;
            default:
                description = 'Custom parameter profile.';
        }
        
        descriptionElement.innerHTML = `<p class="text-muted">${description}</p>`;
        
        // Show profile parameters
        const parameters = this.profiles[selectedProfile];
        if (parameters) {
            const paramList = document.createElement('div');
            paramList.className = 'profile-parameters small';
            
            // Add each parameter
            for (const [key, value] of Object.entries(parameters)) {
                const paramItem = document.createElement('div');
                paramItem.className = 'd-flex justify-content-between';
                paramItem.innerHTML = `
                    <span class="param-name">${this.formatParameterName(key)}:</span>
                    <span class="param-value text-primary">${value}</span>
                `;
                paramList.appendChild(paramItem);
            }
            
            descriptionElement.appendChild(paramList);
        }
    }
    
    /**
     * Format parameter name for display
     * @param {string} name - The parameter name
     * @returns {string} - Formatted name
     */
    formatParameterName(name) {
        // Convert camelCase or snake_case to Title Case
        return name
            .replace(/_/g, ' ')
            .replace(/([A-Z])/g, ' $1')
            .replace(/^./, str => str.toUpperCase());
    }
    
    /**
     * Add event listeners for profile actions
     */
    addProfileEventListeners() {
        // Profile selection change
        const profileSelect = document.getElementById('profileSelect');
        if (profileSelect) {
            profileSelect.addEventListener('change', () => {
                this.currentProfile = profileSelect.value;
                this.updateProfileDescription();
            });
        }
        
        // Apply profile button
        const applyProfileBtn = document.getElementById('applyProfileBtn');
        if (applyProfileBtn) {
            applyProfileBtn.addEventListener('click', () => {
                this.applySelectedProfile();
            });
        }
        
        // Save as new button
        const saveAsNewBtn = document.getElementById('saveAsNewBtn');
        if (saveAsNewBtn) {
            saveAsNewBtn.addEventListener('click', () => {
                this.showSaveProfileModal();
            });
        }
        
        // Save profile button in modal
        const saveProfileBtn = document.getElementById('saveProfileBtn');
        if (saveProfileBtn) {
            saveProfileBtn.addEventListener('click', () => {
                this.saveCustomProfile();
            });
        }
    }
    
    /**
     * Apply the currently selected profile
     */
    applySelectedProfile() {
        const profileSelect = document.getElementById('profileSelect');
        if (!profileSelect) return;
        
        const selectedProfile = profileSelect.value;
        const parameters = this.profiles[selectedProfile];
        
        if (parameters) {
            // Update sliders
            this.updateSliders(parameters);
            
            // Show notification
            this.showNotification(`Applied profile: ${selectedProfile}`);
            
            console.log(`Applied parameter profile: ${selectedProfile}`, parameters);
        }
    }
    
    /**
     * Update all parameter sliders with profile values
     * @param {Object} parameters - The parameters object
     */
    updateSliders(parameters) {
        // Update temperature slider
        const tempSlider = document.getElementById('temperatureSlider');
        if (tempSlider && parameters.temperature !== undefined) {
            tempSlider.value = parameters.temperature;
            // Trigger the input event to update display
            tempSlider.dispatchEvent(new Event('input'));
        }
        
        // Update top_p slider
        const topPSlider = document.getElementById('topPSlider');
        if (topPSlider && parameters.top_p !== undefined) {
            topPSlider.value = parameters.top_p;
            topPSlider.dispatchEvent(new Event('input'));
        }
        
        // Update top_k slider
        const topKSlider = document.getElementById('topKSlider');
        if (topKSlider && parameters.top_k !== undefined) {
            topKSlider.value = parameters.top_k;
            topKSlider.dispatchEvent(new Event('input'));
        }
        
        // Update repetition penalty slider
        const repPenSlider = document.getElementById('repetitionPenaltySlider');
        if (repPenSlider && parameters.repetition_penalty !== undefined) {
            repPenSlider.value = parameters.repetition_penalty;
            repPenSlider.dispatchEvent(new Event('input'));
        }
        
        // Update context window slider
        const contextSlider = document.getElementById('contextWindowSlider');
        if (contextSlider && parameters.context_window !== undefined) {
            contextSlider.value = parameters.context_window;
            contextSlider.dispatchEvent(new Event('input'));
        }
    }
    
    /**
     * Show the save profile modal
     */
    showSaveProfileModal() {
        const modal = document.getElementById('saveProfileModal');
        if (!modal) return;
        
        // Clear the form
        const nameInput = document.getElementById('profileName');
        const descInput = document.getElementById('profileDescription');
        if (nameInput) nameInput.value = '';
        if (descInput) descInput.value = '';
        
        // Set model-specific checkbox to model name
        const modelCheck = document.getElementById('modelSpecificCheck');
        if (modelCheck) {
            modelCheck.checked = false;
            const label = modelCheck.nextElementSibling;
            if (label) {
                label.textContent = `Make this profile specific to ${this.currentModel}`;
            }
        }
        
        // Show the modal
        const modalInstance = new bootstrap.Modal(modal);
        modalInstance.show();
    }
    
    /**
     * Save the current parameters as a new custom profile
     */
    async saveCustomProfile() {
        const nameInput = document.getElementById('profileName');
        const descInput = document.getElementById('profileDescription');
        const modelCheck = document.getElementById('modelSpecificCheck');
        
        if (!nameInput || !nameInput.value.trim()) {
            this.showNotification('Please enter a profile name', 'error');
            return;
        }
        
        // Get current parameters
        const parameters = this.getCurrentParameters();
        
        // Create profile data
        const profileData = {
            name: nameInput.value.trim(),
            description: descInput ? descInput.value.trim() : '',
            parameters: parameters
        };
        
        // Add model specificity if checked
        if (modelCheck && modelCheck.checked) {
            profileData.model = this.currentModel;
        }
        
        try {
            // Save to API
            const response = await fetch('/api/parameter_profiles/custom', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify(profileData)
            });
            
            if (response.ok) {
                const result = await response.json();
                
                // Hide modal
                const modal = document.getElementById('saveProfileModal');
                if (modal) {
                    const modalInstance = bootstrap.Modal.getInstance(modal);
                    if (modalInstance) modalInstance.hide();
                }
                
                // Reload profiles
                await this.loadProfilesForModel(this.currentModel);
                
                // Update dropdown
                this.updateProfilesDropdown();
                
                // Show notification
                this.showNotification(result.message || 'Profile saved successfully');
                
                console.log('Saved custom profile:', profileData);
            } else {
                const result = await response.json();
                this.showNotification(result.message || 'Error saving profile', 'error');
            }
        } catch (error) {
            console.error('Error saving profile:', error);
            this.showNotification('Error saving profile', 'error');
        }
    }
    
    /**
     * Get current parameter values from sliders
     * @returns {Object} - The current parameters
     */
    getCurrentParameters() {
        const parameters = {};
        
        // Get temperature
        const tempSlider = document.getElementById('temperatureSlider');
        if (tempSlider) {
            parameters.temperature = parseFloat(tempSlider.value);
        }
        
        // Get top_p
        const topPSlider = document.getElementById('topPSlider');
        if (topPSlider) {
            parameters.top_p = parseFloat(topPSlider.value);
        }
        
        // Get top_k
        const topKSlider = document.getElementById('topKSlider');
        if (topKSlider) {
            parameters.top_k = parseInt(topKSlider.value);
        }
        
        // Get repetition penalty
        const repPenSlider = document.getElementById('repetitionPenaltySlider');
        if (repPenSlider) {
            parameters.repetition_penalty = parseFloat(repPenSlider.value);
        }
        
        // Get context window
        const contextSlider = document.getElementById('contextWindowSlider');
        if (contextSlider) {
            parameters.context_window = parseInt(contextSlider.value);
        }
        
        return parameters;
    }
    
    /**
     * Show a notification to the user
     * @param {string} message - The message to show
     * @param {string} type - The type of notification (success/error)
     */
    showNotification(message, type = 'success') {
        // Use the global notification system if available
        if (window.showToast) {
            window.showToast(message, type, 3000);
            return;
        }
        
        // Fallback to alert
        alert(message);
    }
}

// Initialize when DOM is loaded
document.addEventListener('DOMContentLoaded', function() {
    // Create global instance
    window.parameterProfiles = new ParameterProfiles();
    
    // Initialize with current model
    const modelSelect = document.getElementById('modelSelect');
    const currentModel = modelSelect ? modelSelect.value : 'llama3.2';
    
    window.parameterProfiles.init(currentModel);
    
    // Update profile when model changes
    if (modelSelect) {
        modelSelect.addEventListener('change', function() {
            window.parameterProfiles.init(this.value);
        });
    }
});